lib/features/feed/cubit/share_feed_post_cubit.dart

```diff
@@ -1,30 +1,104 @@
+import 'package:fluttercon/features/feed/cubit/platform.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
-import 'package:share_plus/share_plus.dart';
 
 part 'share_feed_post_state.dart';
 part 'share_feed_post_cubit.freezed.dart';
 
 class ShareFeedPostCubit extends Cubit<ShareFeedPostState> {
   ShareFeedPostCubit() : super(const ShareFeedPostState.initial());
 
-  Future<void> sharePost(Feed feed) async {
+  final AppinioSocialShare appinioSocialShare = AppinioSocialShare();
+
+  Future<void> sharePost(Feed feed, Platform platform) async {
     emit(const ShareFeedPostState.loading());
+
     try {
+      final message = '${feed.body}\n\nhttps://fluttercon.dev';
+      String? filePath;
+
       if (feed.image != null) {
         final files = await Misc.downloadFiles([feed.image!]);
-        await Share.shareXFiles(
-          files,
-          text: '${feed.body}\n\nhttps://fluttercon.dev',
-        );
-      } else {
-        await Share.share('${feed.body}\n\nhttps://fluttercon.dev');
+        if (files.isNotEmpty) {
+          filePath = files.first.path;
+        }
       }
+
+      final installedApps = await appinioSocialShare.getInstalledApps();
+
+      switch (platform) {
+        case Platform.telegram:
+          if (installedApps['telegram'] ?? false) {
+            await _shareToTelegram(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('Telegram is not installed.'));
+          }
+        case Platform.whatsapp:
+          if (installedApps['whatsapp'] ?? false) {
+            await _shareToWhatsApp(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('WhatsApp is not installed.'));
+          }
+        case Platform.twitter:
+          if (installedApps['twitter'] ?? false) {
+            await _shareToTwitter(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('Twitter is not installed.'));
+          }
+        case Platform.facebook:
+          if (installedApps['facebook'] ?? false) {
+            if (filePath == null) {
+              emit(
+                const ShareFeedPostState.error(
+                  'Failed to share post. Kindly share a post with an image.',
+                ),
+              );
+            } else {
+              await _shareToFacebook(message, filePath);
+            }
+          } else {
+            emit(const ShareFeedPostState.error('Facebook is not installed.'));
+          }
+      }
+
       emit(const ShareFeedPostState.loaded());
     } catch (e) {
-      emit(ShareFeedPostState.error(e.toString()));
+      emit(ShareFeedPostState.error('Failed to share post: $e'));
+    }
+  }
+
+  Future<void> _shareToTelegram(String message, String? filePath) async {
+    try {
+      await appinioSocialShare.android.shareToTelegram(message, filePath);
+    } catch (e) {
+      throw Exception('Error sharing to Telegram: $e');
```

Comment:
Throw `Failure` instead. 

Sample: `lib/common/utils/network.dart:75`

Then in the calling function, catch it like so `lib/features/about/cubit/fetch_individual_organisers_cubit.dart:32-34`

Reply:
On second thought, you don't have to actually throw anything here. Let me pull your branch and make suggestions after I have seen the code on my local.